plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftClient.java

```diff
@@ -25,43 +26,85 @@
 import static io.trino.plugin.jdbc.StandardColumnMappings.realColumnMapping;
+import static io.trino.plugin.jdbc.StandardColumnMappings.realWriteFunction;
+import static io.trino.plugin.jdbc.StandardColumnMappings.shortDecimalWriteFunction;
 import static io.trino.plugin.jdbc.StandardColumnMappings.smallintColumnMapping;
+import static io.trino.plugin.jdbc.StandardColumnMappings.smallintWriteFunction;
 import static io.trino.plugin.jdbc.StandardColumnMappings.timeColumnMappingUsingSqlTime;
 import static io.trino.plugin.jdbc.StandardColumnMappings.timestampColumnMappingUsingSqlTimestampWithRounding;
 import static io.trino.plugin.jdbc.StandardColumnMappings.tinyintColumnMapping;
+import static io.trino.plugin.jdbc.StandardColumnMappings.tinyintWriteFunction;
 import static io.trino.plugin.jdbc.StandardColumnMappings.varbinaryColumnMapping;
+import static io.trino.plugin.jdbc.StandardColumnMappings.varbinaryWriteFunction;
+import static io.trino.plugin.jdbc.StandardColumnMappings.varcharWriteFunction;
 import static io.trino.plugin.jdbc.TypeHandlingJdbcSessionProperties.getUnsupportedTypeHandling;
 import static io.trino.plugin.jdbc.UnsupportedTypeHandling.CONVERT_TO_VARCHAR;
 import static io.trino.spi.StandardErrorCode.NOT_SUPPORTED;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.BooleanType.BOOLEAN;
+import static io.trino.spi.type.DateType.DATE;
 import static io.trino.spi.type.DecimalType.createDecimalType;
+import static io.trino.spi.type.DoubleType.DOUBLE;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.spi.type.RealType.REAL;
+import static io.trino.spi.type.SmallintType.SMALLINT;
 import static io.trino.spi.type.TimestampType.TIMESTAMP_MILLIS;
+import static io.trino.spi.type.TinyintType.TINYINT;
+import static io.trino.spi.type.VarbinaryType.VARBINARY;
 import static java.lang.Math.max;
 import static java.lang.String.format;
 
 public class RedshiftClient
         extends BaseJdbcClient
 {
+    /**
+     * @deprecated To be removed after {{@link #legacyToWriteMapping} is removed.
+     */
+    @Deprecated
+    private static final Map<Type, WriteMapping> WRITE_MAPPINGS = ImmutableMap.<Type, WriteMapping>builder()
+            .put(BOOLEAN, WriteMapping.booleanMapping("boolean", booleanWriteFunction()))
+            .put(BIGINT, WriteMapping.longMapping("bigint", bigintWriteFunction()))
+            .put(INTEGER, WriteMapping.longMapping("integer", integerWriteFunction()))
+            .put(SMALLINT, WriteMapping.longMapping("smallint", smallintWriteFunction()))
+            .put(TINYINT, WriteMapping.longMapping("tinyint", tinyintWriteFunction()))
+            .put(DOUBLE, WriteMapping.doubleMapping("double precision", doubleWriteFunction()))
+            .put(REAL, WriteMapping.longMapping("real", realWriteFunction()))
+            .put(VARBINARY, WriteMapping.sliceMapping("varbinary", varbinaryWriteFunction()))
```

Comment:
this is misleading. it looks like we know what's the proper mapping for redshift, whereas we do not

Reply:
I thought below comment is sufficient. I will extract a method and move these mapping to there if needed. 
```
L161. // TODO (https://github.com/trinodb/trino/issues/497) Implement proper type mapping and add test
```